src/DeterministicNoiseSimulator.cpp

```diff
@@ -308,7 +368,7 @@ void DeterministicNoiseSimulator::ApplyAmplitudeDampingToNode(std::array<qc::Mat
             CN::mul(helper_edge[0].w, complex_prob, e[3].w);
             helper_edge[0].p = e[3].p;
             dd::Edge tmp     = dd->add2(e[0], helper_edge[0]);
-            if (!e[0].w.approximatelyZero()) {
+            if (e[0].w != dd::Complex::zero) {
```

Comment:
try to be consistent with these checks. in the functions below approximate comparison is used for practically all checks.

Reply:
`e[0].w.approximatelyZero()` and `dd::Complex::zero` behave differently and the usage is intend. I.e., I use `dd::Complex::zero` to check for the static entry zero (which should not be put into the cache table).